central/reports/service/v2/service_impl.go

```diff
@@ -0,0 +1,93 @@
+package v2
+
+import (
+	"context"
+
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/pkg/errors"
+	metadataDS "github.com/stackrox/rox/central/reports/metadata/datastore"
+	"github.com/stackrox/rox/central/role/resources"
+	apiV2 "github.com/stackrox/rox/generated/api/v2"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/auth/permissions"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/grpc/authz"
+	"github.com/stackrox/rox/pkg/grpc/authz/perrpc"
+	"github.com/stackrox/rox/pkg/grpc/authz/user"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/search"
+	"google.golang.org/grpc"
+)
+
+var (
+	log = logging.LoggerForModule()
+
+	authorizer = perrpc.FromMap(map[authz.Authorizer][]string{
+		user.With(permissions.View(resources.WorkflowAdministration)): {
+			"/v2.ReportService/GetReportStatus",
+			"/v2.ReportService/GetLastReportStatusConfigID",
+		},
+	})
+)
+
+type serviceImpl struct {
+	apiV2.UnimplementedReportServiceServer
+	metadataDatastore metadataDS.DataStore
+}
+
+func (s *serviceImpl) RegisterServiceServer(grpcServer *grpc.Server) {
+	if features.VulnMgmtReportingEnhancements.Enabled() {
```

Comment:
A note that you do need to guard the registration behind the feature flag. Putting the calls to `Singleton()` behind the feature flag skips the service altogether.

Reply:
I think you meant "you *do not* need to guard..."

Reply:
Yes, I updated my comment